api/src/controllers/verification.ts

```diff
@@ -40,15 +74,26 @@ export const formVerification = async (req: AppRequest<ManualContractVerificatio
     res.send('Verified');
   } catch (err) {
     if (err.status === 404) {
-      await contractVerificationRequestInsert({
-        ...req.body, success: false, optimization: req.body.optimization === 'true', args: req.body.arguments, errorMessage: err.message,
-      });
+      await Promise.resolve()
+        .then(() => ensureVerificationRequest(req.body))
```

Comment:
it seems weird to be that we are validating the request body after failure.  Is there any reason why we aren't validating the request as the first step in the api request? Maybe we could push `ensureObjectKeys(req.body, [...requiredArguments, 'token']);` into `ensureVerificationRequest(req.body)` and then call  `ensureVerificationRequest(req.body)` as the first statement in the `try` body above?

Reply:
In addition it would probably be nicer if we were using some form of request validation library to validate requests.  See this example for schema definition https://github.com/paralink-network/paralink-node/blob/master/src/pql/schema.py and then application https://github.com/paralink-network/paralink-node/blob/master/src/pql/parser.py#L29-L41